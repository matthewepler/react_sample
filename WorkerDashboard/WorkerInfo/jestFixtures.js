import { fromJS } from 'immutable'

// @MATEO - these tests will break if the data format changes. Is there a better
// way to do this?

export const currWorker = fromJS({     // pasted from API response to https://hcs-ace-api.herokuapp.com/clients/27/projects/37/groups/65/users/129
  lastname: 'Leader',
  gender: null,
  unionStatus: null,
  completedRegistration: false,
  resourceId: null,
  userStatuses: [
    {
      groupId: 65,
      projectId: 37,
      userId: 129,
      status: 'inactive',
      createdAt: '2017-06-14T14:56:16.948Z',
      updatedAt: '2017-06-14T14:56:16.948Z',
      project: {
        userId: 129,
        clientId: 27,
        resourceId: 689,
        endDate: '2020-06-10T16:00:00.000Z',
        city: 'New York',
        name: 'Sample Project 1',
        zipCode: '10006',
        updatedAt: '2017-06-14T14:56:16.882Z',
        startDate: '2017-06-14T16:00:00.000Z',
        state: 'NY',
        timeSheetGenerationDay: null,
        videoId: null,
        timeZone: 'America/New_York',
        industryId: 2,
        id: 37,
        createdAt: '2017-06-14T14:56:16.882Z',
        industry: {
          id: 2,
          name: 'Industry Two',
          createdAt: '2017-05-09T12:36:54.000Z',
          updatedAt: '2017-05-09T12:36:54.000Z'
        }
      },
      group: {
        userId: 129,
        userGroupRoles: [
          {
            endDate: '2020-06-10T16:00:00.000Z',
            roleName: 'Group Leader',
            roleId: 1,
            userId: 129,
            groupId: 65,
            startDate: '2017-06-14T16:00:00.000Z',
            createdAt: '2017-06-14T14:56:16.935Z',
            updatedAt: '2017-06-14T14:56:16.935Z'
          }
        ],
        city: null,
        name: 'Project Leaders',
        zipCode: null,
        updatedAt: '2017-06-14T14:56:16.908Z',
        shifts: null,
        state: null,
        address: null,
        id: 65,
        createdAt: '2017-06-14T14:56:16.908Z',
        tradeId: 34
      }
    }
  ],
  city: null,
  zipCode: null,
  updatedAt: '2017-06-14T14:56:16.839Z',
  deactivated: false,
  state: null,
  streetAddress: null,
  yearsOfExperience: null,
  phoneNumber: '+1 212-333-3333',
  language: 'english',
  dateOfBirth: null,
  id: 129,
  createdAt: '2017-06-14T14:56:16.839Z',
  firstname: 'Fearless',
  receiveReports: {
    dailyReports: 'true',
    weeklyReports: 'true'
  },
  email: 'test@gmail.com',
  confirmed: false
})

// @MATEO - these tests will break if the data format changes. Is there a better
// way to do this?

export const trades = fromJS({
  count: 2,
  rows: [
    {
      name: 'Boilermaker',
      id: 1,
      createdAt: '2017-05-09T12:36:54.000Z',
      updatedAt: '2017-05-09T12:36:54.000Z'
    },
    {
      name: 'Carpenter',
      id: 2,
      createdAt: '2017-05-09T12:36:54.000Z',
      updatedAt: '2017-05-09T12:36:54.000Z'
    }
  ]
})

// @MATEO - these tests will break if the data format changes. Is there a better
// way to do this?

export const projectGroups = fromJS([
  {
    user: {
      firstname: 'first1',
      lastname: 'last1',
      userStatuses: [{
        groupId: 11,
        userId: 111
      }]
    }
  },
  {
    user: {
      firstname: 'first2',
      lastname: 'last2',
      userStatuses: [{
        groupId: 22,
        userId: 222
      }]
    }
  }
])