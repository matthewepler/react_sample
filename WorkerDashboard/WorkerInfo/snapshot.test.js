import React from 'react'
import { mount, render, shallow } from 'enzyme'
// import { renderer } from 'react-test-renderer'
import { fromJS } from 'immutable'
import { StyleSheetTestUtils } from 'aphrodite/no-important'
import { Component as WorkerInfo } from './'
import { currWorker, trades, projectGroups } from './jestFixtures'

let routerMock

describe('WorkerInfo container snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection()
    routerMock = resetMockRouter()
  })
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
    jest.resetAllMocks()
  })

  it('renders WorkerInfo component snapshot', () => {
    const tree = renderComponent()
    expect(tree).toMatchSnapshot()
  })

  it('renders worker data section when passed currWorker and trades data via props', () => {
    const tree = renderComponent()
    expect(tree.find(sel('worker-data-wrapper'))).toHaveLength(0)
    tree.setProps({ currWorker, trades })
    expect(tree.find(sel('worker-data-wrapper'))).toHaveLength(1)
  })

  it('renders the passed prop data for a worker correctly in the DOM', () => {
    const workerName = `${currWorker.get('firstname')} ${currWorker.get('lastname')}`
    const tree = renderComponent()
    tree.setProps({ currWorker, trades })
    expect(tree.find(sel('worker-name')).text()).toEqual(workerName)
  })

  it('does not render worker data that is null', () => {
    const tree = renderComponent()
    tree.setProps({ currWorker, trades })
    expect(tree.find(sel('worker-email'))).toHaveLength(1)

    let currWorkerMod = currWorker.merge(fromJS({ email: null }))
    tree.setProps({ currWorker: currWorkerMod })
    expect(tree.find(sel('worker-email'))).toHaveLength(0)
  })

  it('renders a list of workers with projectGroups prop', () => {
    const tree = renderComponent()
    tree.setProps({ currWorker, trades })
    tree.instance().createWorkerList(projectGroups)
    tree.instance().filterWorkerList()
    expect(tree.find(sel('search-results-li'))).toHaveLength(2)
  })

  it('filters search results based on this.state.searchStr', () => {
    const tree = renderComponent()
    tree.setProps({ currWorker, trades })
    tree.instance().createWorkerList(projectGroups)
    tree.setState({ searchStr: '2' })
    tree.instance().filterWorkerList()
    expect(tree.find(sel('search-results-li'))).toHaveLength(1)
    expect(tree.find(sel('search-results-li')).text()).toBe(
      `${projectGroups.get(1).getIn(['user', 'firstname'])} ${projectGroups.get(1).getIn(['user', 'lastname'])}`
    )
  })

  it('trigger fetch for new worker data when a search result is clicked', () => {
    const tree = renderComponent()
    tree.setProps({ currWorker, trades })
    tree.instance().createWorkerList(projectGroups)
    tree.instance().filterWorkerList()
    tree.instance().handleSearchResultClick = jest.fn()

    const firstResult = tree.find(sel('search-results-li')).first()
    firstResult.simulate('click')

    const resultsUser = projectGroups.first().getIn(['user', 'userStatuses']).first()

    expect(tree.instance().handleSearchResultClick).toHaveBeenCalledWith(
      resultsUser.get('userId'), resultsUser.get('groupId')
    )
  })
})

function renderComponent (props = {}) {
  const propsToUse = {
    router: routerMock,
    tradesFetch: () => {},
    workerInfoFetch: () => {},
    ...props
  }
  return shallow(<WorkerInfo {...propsToUse} />)
  // not passing anything via store because the component should still render
  // without it. instead, using setProps() to update component, mocking new data
  // from the parent <Dashboard /> component, passed when a new fetch is completed.
}

function sel (id) {
  return `[data-test="${id}"]`
}

function resetMockRouter () {
  return {
    params: { id: 1, clientId: 1 }
  }
}
