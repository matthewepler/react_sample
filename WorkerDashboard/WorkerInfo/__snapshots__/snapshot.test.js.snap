exports[`WorkerInfo component snapshot renders WorkerInfo component snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    sizeHoriz="small"
    title=""
    type="default">
    <div
        className="noOverflowX_1hsn6c7">
        <SearchButton
            onBlurHandler={[Function]}
            onChangeHandler={[Function]}
            onFocusHandler={[Function]}
            onIconClick={[Function]}
            onSubmitHandler={[Function]} />
        <AlertButton />
        <div
            className="loadingWrapper_6tm9cs">
            <p>
                Loading...
            </p>
        </div>
    </div>
</Card>,
  "nodes": Array [
    <Card
      sizeHoriz="small"
      title=""
      type="default">
      <div
            className="noOverflowX_1hsn6c7">
            <SearchButton
                  onBlurHandler={[Function]}
                  onChangeHandler={[Function]}
                  onFocusHandler={[Function]}
                  onIconClick={[Function]}
                  onSubmitHandler={[Function]} />
            <AlertButton />
            <div
                  className="loadingWrapper_6tm9cs">
                  <p>
                        Loading...
                  </p>
            </div>
      </div>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <WorkerInfo
        router={
                Object {
                        "params": Object {
                          "clientId": 1,
                          "id": 1,
                        },
                      }
        }
        tradesFetch={[Function]}
        workerInfoFetch={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WorkerInfo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "filterFocus": false,
        "params": null,
        "props": Object {
          "router": Object {
            "params": Object {
              "clientId": 1,
              "id": 1,
            },
          },
          "tradesFetch": [Function],
          "workerInfoFetch": [Function],
        },
        "refs": Object {},
        "state": Object {
          "groupFilter": null,
          "groupFilterList": null,
          "searchResults": Array [],
          "searchStr": "",
          "workerData": null,
          "workerList": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          sizeHoriz="small"
          title=""
          type="default">
          <div
                    className="noOverflowX_1hsn6c7">
                    <SearchButton
                              onBlurHandler={[Function]}
                              onChangeHandler={[Function]}
                              onFocusHandler={[Function]}
                              onIconClick={[Function]}
                              onSubmitHandler={[Function]} />
                    <AlertButton />
                    <div
                              className="loadingWrapper_6tm9cs">
                              <p>
                                        Loading...
                              </p>
                    </div>
          </div>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card
          sizeHoriz="small"
          title=""
          type="default">
          <div
                    className="noOverflowX_1hsn6c7">
                    <SearchButton
                              onBlurHandler={[Function]}
                              onChangeHandler={[Function]}
                              onFocusHandler={[Function]}
                              onIconClick={[Function]}
                              onSubmitHandler={[Function]} />
                    <AlertButton />
                    <div
                              className="loadingWrapper_6tm9cs">
                              <p>
                                        Loading...
                              </p>
                    </div>
          </div>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <WorkerInfo
    router={
        Object {
            "params": Object {
              "clientId": 1,
              "id": 1,
            },
          }
    }
    tradesFetch={[Function]}
    workerInfoFetch={[Function]} />,
}
`;

exports[`WorkerInfo container snapshot renders WorkerInfo component snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    sizeHoriz="small"
    title=""
    type="default">
    <div
        className="noOverflowX_1hsn6c7">
        <SearchButton
            onBlurHandler={[Function]}
            onChangeHandler={[Function]}
            onFocusHandler={[Function]}
            onIconClick={[Function]}
            onSubmitHandler={[Function]} />
        <AlertButton />
        <div
            className="loadingWrapper_6tm9cs">
            <p>
                Loading...
            </p>
        </div>
    </div>
</Card>,
  "nodes": Array [
    <Card
      sizeHoriz="small"
      title=""
      type="default">
      <div
            className="noOverflowX_1hsn6c7">
            <SearchButton
                  onBlurHandler={[Function]}
                  onChangeHandler={[Function]}
                  onFocusHandler={[Function]}
                  onIconClick={[Function]}
                  onSubmitHandler={[Function]} />
            <AlertButton />
            <div
                  className="loadingWrapper_6tm9cs">
                  <p>
                        Loading...
                  </p>
            </div>
      </div>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <WorkerInfo
        router={
                Object {
                        "params": Object {
                          "clientId": 1,
                          "id": 1,
                        },
                      }
        }
        tradesFetch={[Function]}
        workerInfoFetch={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WorkerInfo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "filterFocus": false,
        "params": null,
        "props": Object {
          "router": Object {
            "params": Object {
              "clientId": 1,
              "id": 1,
            },
          },
          "tradesFetch": [Function],
          "workerInfoFetch": [Function],
        },
        "refs": Object {},
        "state": Object {
          "groupFilter": null,
          "groupFilterList": null,
          "searchResults": Array [],
          "searchStr": "",
          "workerData": null,
          "workerList": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          sizeHoriz="small"
          title=""
          type="default">
          <div
                    className="noOverflowX_1hsn6c7">
                    <SearchButton
                              onBlurHandler={[Function]}
                              onChangeHandler={[Function]}
                              onFocusHandler={[Function]}
                              onIconClick={[Function]}
                              onSubmitHandler={[Function]} />
                    <AlertButton />
                    <div
                              className="loadingWrapper_6tm9cs">
                              <p>
                                        Loading...
                              </p>
                    </div>
          </div>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card
          sizeHoriz="small"
          title=""
          type="default">
          <div
                    className="noOverflowX_1hsn6c7">
                    <SearchButton
                              onBlurHandler={[Function]}
                              onChangeHandler={[Function]}
                              onFocusHandler={[Function]}
                              onIconClick={[Function]}
                              onSubmitHandler={[Function]} />
                    <AlertButton />
                    <div
                              className="loadingWrapper_6tm9cs">
                              <p>
                                        Loading...
                              </p>
                    </div>
          </div>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <WorkerInfo
    router={
        Object {
            "params": Object {
              "clientId": 1,
              "id": 1,
            },
          }
    }
    tradesFetch={[Function]}
    workerInfoFetch={[Function]} />,
}
`;
